Deepseek

barci2 opened the pull request:
### Ticket
Initial skeleton for the DeepSeek bringup

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] Tests for Embedding1D plugged into CI 
- [ ] Squash commits

yieldthought committed:
#23022: Add reference code

yieldthought committed:
#23022: Extract reference code with opus4, not hand-verified yet!

yieldthought committed:
#0: Define submodule structure with MLP_1D example from TTT

yieldthought committed:
#23022: More Submodule definition

yieldthought committed:
#23022: Add more example classes

yieldthought committed:
#23022: Remove moved files

yieldthought committed:
#23022: Use dataclass, add embedding

yieldthought committed:
#23022: Update README

yieldthought committed:
#23022: Clean up reference folder

yieldthought committed:
#23022: Fix reference models

barci2 committed:
Clean up config-handling code

Signed-off-by: Bartłomiej Cieślar <bcieslar@tenstorrent.com>

barci2 committed:
Bugfixes

Signed-off-by: Bartłomiej Cieślar <bcieslar@tenstorrent.com>

barci2 committed:
Update RunConfig to be created by a class and contain the mesh device

Signed-off-by: Bartłomiej Cieślar <bcieslar@tenstorrent.com>

barci2 committed:
Example MLP1D impl

Signed-off-by: Bartłomiej Cieślar <bcieslar@tenstorrent.com>

barci2 committed:
Introduce custom state to the Module and add documentation

Signed-off-by: Bartłomiej Cieślar <bcieslar@tenstorrent.com>

yieldthought committed:
#23022: Fix some of the bugs in the framework

barci2 committed:
Interface changes

yieldthought committed:
#23022: Fix bugs in framework, update README

yieldthought committed:
#23022: Apply changes to MLP, prefill test passes, decode test needs work

pprajapatiTT committed:
Add support for MOE expert

yieldthought committed:
#23022: Update codeowners for deepseek project

yieldthought committed:
#23022: Fix MLP sharding, decode now passes for 1 chip

yieldthought committed:
Merge branch 'main' into deepseek

yieldthought committed:
#23022: Remove CODEOWNERS change so we can merge deepseek

yieldthought committed:
#23022: Pass copyright hooks

yieldthought committed:
Merge branch 'main' into deepseek

yieldthought committed:
#23022: Pass CODEOWNERS validator

nsextonTT reviewed (dismissed):
Your CODEOWNERS file is completely messed up. If you are trying to get it to match or sync with main, make sure you are checking that file out specifying the remote and not your local main.

pprajapatiTT committed:
Fix Expert layer of MOE. Decode working with PCC=0.99xx

yieldthought committed:
#23022: CODEOWNERS from remote main

yieldthought committed:
#23022: Update copyright tags

yieldthought commented on models/demos/deepseek_v3/tt/embedding_1d.py:
```diff
@@ -0,0 +1,121 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
```
Updated them all

yieldthought committed:
#23022: Add test for nested modules

kpaigwar committed:
Deepseek rms norm (#24558)

### Problem description
Bringup RMSNorm for
1. Decoder (Prefill/Decode)
2. MLA (Prefill/Decode)

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI with demo tests passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml)
CI passes (if applicable) See [recommended dev
flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] New/Existing tests provide coverage for changes

yieldthought committed:
#23022: Use common file for rmsnorm reference code

yieldthought committed:
#23022: Update copyright tags

yieldthought committed:
#23022: Precommit hook passes
